Use the model that we trained previously in Lab 24 to make predictions on the charges of these clients with the following information:

Sheldon Cooper:
* Age = 30
* Sex = Male
* BMI = 28
* Children = 0
* Smoker = no
* Region = southwest

Lily Potter:
* Age = 27
* Sex =Female
* BMI = 22
* Children = 1
* Smoker = no
* Region = southeast

Morticia Addams:
* Age = 50
* Sex =Female
* BMI = 24
* Children = 2
* Smoker = yes
* Region = southwest<br>
For consistence, we can use the codes I have below. Item 6 is just added.

#6. Predictions on specific clients

Sheldon Cooper:
* Age = 30
* Sex = Male
* BMI = 28
* Children = 0
* Smoker = no
* Region = southwest

Lily Potter:
* Age = 27
* Sex =Female
* BMI = 22
* Children = 1
* Smoker = no
* Region = southeast

Morticia Addams:
* Age = 50
* Sex =Female
* BMI = 24
* Children = 2
* Smoker = yes
* Region = southwest

###Upload and Load model from Lab 24.

In [ ]:
# saved in C:\Users\klbai\OneDrive\Desktop\ADA\trained_model.joblib
from google.colab import files

# Upload 'trained_model.joblib' to Colab
uploaded = files.upload()

Saving trained_model.joblib to trained_model.joblib


In [ ]:
from joblib import load

# Specify the path  'trained_model.joblib' in Colab
model_path = 'trained_model.joblib'

# Load the saved model
model = load(model_path)

###Create a DataFrame with the client information

In [ ]:
import pandas as pd

# Define the client information
client_data = {
    'age': [30, 27, 50],
    'sex': ['male', 'female', 'female'],
    'bmi': [28, 22, 24],
    'children': [0, 1, 2],
    'smoker': ['no', 'no', 'yes'],
    'region': ['southwest', 'southeast', 'southwest']
}

# Create a DataFrame from the client data
client_df = pd.DataFrame(client_data)


In [ ]:
client_df #(0=Sheldon Cooper, 1=Lily Potter, 2=Morticia Addams)

age     sex  bmi  children smoker     region
0   30    male   28         0     no  southwest
1   27  female   22         1     no  southeast
2   50  female   24         2    yes  southwest

##Datapreprocessing

In [ ]:
from sklearn.preprocessing import LabelEncoder

# Initialize the LabelEncoder
labelenc = LabelEncoder()

# Apply LabelEncoder to categorical columns and print classes in for loop
for col in ['sex', 'smoker', 'region']:
    client_df[col] = labelenc.fit_transform(client_df[col])
    print(f"Classes for '{col}': {labelenc.classes_}")

print()

# Display the updated DataFrame
client_df

# female=0, male=0; not smoker=0, smoker=1; region 0=southeast, 1=southwest (alphabetically)

Classes for 'sex': ['female' 'male']
Classes for 'smoker': ['no' 'yes']
Classes for 'region': ['southeast' 'southwest']



age  sex  bmi  children  smoker  region
0   30    1   28         0       0       1
1   27    0   22         1       0       0
2   50    0   24         2       1       1

##Make Predictions

Prepare selected features

In [ ]:
# selected_feature_indices is defined from previous feature selection step
selected_feature_indices = [0, 2, 4]  # Determined in Lab 24

# Select only the relevant features based on selected_feature_indices
client_df_selected = client_df.iloc[:, selected_feature_indices]


Predictions

In [ ]:
# Make predictions on client data
client_predictions = model.predict(client_df_selected)

# Print predictions
print("Predicted Charges for Clients:")
for prediction in client_predictions:
    print(prediction)


Predicted Charges for Clients:
5136.46438935265
2325.2300732318126
32507.00862923207


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:432: UserWarning: X has feature names, but LinearRegression was fitted without feature names
  warnings.warn(


Model's estimate on new client's charges are:

0=Sheldon Cooper: 5136.46

1=Lily Potter: 2325.23

2=Morticia Addams: 32507.01

Summary: As long as client_df in Lab 26 is structured and preprocessed identically to df in Lab 24, the predictions obtained are accurate and can be relied upon for making estimations about charges for the new clients.